apps/parse/lib/parse/calendar_attributes.ex

```diff
@@ -0,0 +1,40 @@
+defmodule Parse.CalendarAttributes do
+  @moduledoc """
+  Parser for GTFS calendar_attributes.txt
+  """
+
+  use Parse.Simple
+
+  @type t :: %__MODULE__{
+          service_id: String.t(),
+          description: String.t() | nil,
+          schedule_name: String.t() | nil,
+          schedule_type: String.t() | nil,
+          schedule_typicality: integer() | nil
+        }
+
+  defstruct [
+    :service_id,
+    :description,
+    :schedule_name,
+    :schedule_type,
+    :schedule_typicality
```

Comment:
You can avoid the `|| 0` below by making 0 the default here, and the `copy_int` default as well.

Reply:
In the current way there's only one place in charge of creating the default.

Reply:
That's true, but it's not near the parsing code at all. In the future, you can do something like this:
```elixir
@default_typicality 0

defstruct [typicality: @default_typicality]

def copy_int(""), do: @default_typicality
def copy_int(binary), do: String.to_integer(binary)
```